velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -223,11 +251,39 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VELOX_USER_CHECK(dataToWrite->type()->equivalent(*schema_));
+  auto rowVector =
+      std::dynamic_pointer_cast<facebook::velox::RowVector>(dataToWrite);
+  VELOX_CHECK_NULL(rowVector->nulls());
+
+  // Update the shcema for the complex type.
+  auto childrens = rowVector->children();
+  const auto size = schema_->size();
+  std::vector<VectorPtr> newChildrens = childrens;
+  for (auto i = 0; i < size; i++) {
+    // Only complex type need this copy.
```

Comment:
This copy is expensive. Why do we need this? Also, this change is not described in the PR description.

Reply:
@mbasmanova The RowVector type is defined as `ROW({"c0", "c1"}, {INTEGER(), ROW({"c2"}, {INTEGER()})})`.
The newType is defined as `ROW({"int32", "struct"}, {INTEGER(), ROW({"struct_c0"}, {INTEGER()})})`. For this nested type, if we do not modify the type name inside the 'children' by copying it here, the resulting file schema will be: `ROW({"int32", "struct"}, {INTEGER(), ROW({"c2"}, {INTEGER()})})`.

Reply:
I see... but it seems wasteful to copy all the data just for the purpose of specifying the schema. Let's find a better way.

Reply:
@mbasmanova Added` setType()` method in `BaseVector.h` to avoid the copy operator.  Can you help to review again?